spec/Records.tex

```diff
@@ -677,56 +548,12 @@ \subsection{Arguments}
 \index{classes!arguments}
 \index{records!arguments}
 
-The semantics of argument passing is determined by the type of the formal
-argument (as declared inside the function header).  An actual argument is of a
-type compatible with the formal argument only if it is legal to assign the
-actual to the formal.
-%REVIEW: vass: how is this different from the general rule for argument passing?
-
-%REVIEW: vass: only the implicit conversion rules are different.
-% Argument passing is the same.
-Specifically, if the formal argument is of class type, the actual argument must
-be of that class type or of a type derived from that class type.  If the formal
-argument is of a record type, then it is only necessary for the fields in the
-actual argument to ``cover'' the fields in the formal argument type.
-
-By default, the receiver argument is passed by value for class methods but is
-passed by \chpl{ref} or \chpl{const ref} for record methods.
-In both cases modifications to the receiver fields are visible
-outside the method.
-
-\subsection{Inheritance}
-\label{Inheritance_Differences}
-\index{records!inheritance}
-\index{inheritance!records}
-\index{records!inheritance}
-\index{inheritance!records}
-
-The difference between record inheritance and class
-inheritance is that for records there is no dynamic dispatch.  The record type of
-a variable is the exact type of that variable, i.e. a variable of a
-base record type cannot store a derived record type.
-
-Casting a derived record type to a base record type truncates all 
-fields except those belonging to the base record type.  In the same way, only
-those methods accessible to the base record type may be invoked using the result
-of such a cast.
-
-\subsection{Shadowing and Overriding}
-\label{Base_Method_Differences}
-\index{classes!shadowing}
-\index{records!shadowing}
-\index{classes!overriding}
-\index{records!overriding}
-
-Class variables have run-time types and (therefore) support dynamic dispatch.
-Records are statically typed, so they do not have run-time types and they do not
-support dynamic dispatch.
-
-As a result, in record type hierarchies, shadowing and overriding are the same.  
-Which field is accessed
-and/or which method is invoked is determined statically by the declared type of
-the record being referenced.
+Record arguments use the \chpl{const ref} intent by default - in contrast
+to class arguments which pass by \chpl{in} intent by default.
```

Comment:
It's `const in` for classes.

Ditto below for the receiver - it gotta be `const in` as well.